web/event_handler.go

```diff
@@ -218,36 +218,13 @@ func (fh *eventHandler) handleDeploymentStepSuccess(body []byte) error {
 }
 
 func (fh *eventHandler) deregisterAllAppServices(app *apps.App) []error {
-
 	errors := []error{}
-	serviceName := fh.serviceRegistry.ServiceName(app)
-
-	log.WithField("AppId", app.ID).WithField("ServiceName", serviceName).Info("Deregistering all services")
-
-	services, err := fh.serviceRegistry.GetServices(serviceName)
-
-	if err != nil {
-		log.WithField("Id", app.ID).WithError(err).Error("There was a problem getting Consul services")
-		errors = append(errors, err)
-		return errors
-	}
-
-	if len(services) == 0 {
-		log.WithField("AppId", app.ID).WithField("ServiceName", serviceName).Info("No matching Consul services found")
-		return errors
-	}
+	log.WithField("AppId", app.ID).Info("Deregistering all services")
 
-	for _, service := range services {
-		taskId, err := service.TaskId()
+	for _, task := range app.Tasks {
```

Comment:
`TestWebHandler_HandleDeploymentInfoWithStopApplicationAction` looks like a test that should fail here, but it might be wrong (generated json request body).